File: pybot/wallet.py
"""Wallet loading — reads Solana keypair from JSON file."""

import json
from pathlib import Path

from solders.keypair import Keypair


def load_keypair(wallet_path: str) -> Keypair:
    resolved = Path(wallet_path).expanduser().resolve()
    if not resolved.exists():
        raise FileNotFoundError(f"Wallet file not found: {resolved}")

    secret = json.loads(resolved.read_text())
    return Keypair.from_bytes(bytes(secret))

File: bot/src/utils/wallet.ts
import { Keypair } from "@solana/web3.js";
import fs from "fs";
import path from "path";

export function loadKeypair(walletPath: string): Keypair {
  const resolved = walletPath.replace(/^~/, process.env.HOME!);
  const abs = path.resolve(resolved);

  if (!fs.existsSync(abs)) {
    throw new Error(`Wallet file not found: ${abs}`);
  }

  const secret = JSON.parse(fs.readFileSync(abs, "utf-8"));
  return Keypair.fromSecretKey(Uint8Array.from(secret));
}
